### PR TITLE
[core] Refactor charutils::AvatarPerpetuationReduction

### LINKED ISSUE #778: Summoner's Doublet not working
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened
- [x] I have checked the commit log to see if the issue has been resolved since my server was last updated
- [x] I have read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I have specified what branch this happens on branch: everything

**_Additional Information_** (Steps to reproduce/Expected behavior) **:** 

`Depending on day: Avatar perpetuation cost -3` is currently not working on Summoner's Doublet and probably its variants.

Tested with Garuda on Windsday.

Abbvreviated name: `summoners_dblt.`
Item ID: `15101`

https://www.bg-wiki.com/ffxi/Summoner's_Dblt.

### PR DESCRIPTION
**Pull request by zach2good**
Fixes https://github.com/LandSandBoat/server/issues/778
It's a little confusing in there, there was an off-by-one for the DAY_REDUCTION mod lookup. 
I went through the affinity stuff, that all seems OK.
Don't really understand whats going on with the WEATHER_REDUCTION thing.

**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

### CODE REVIEW

**Comment by zach2good** [comment]
Tested on stream:
Summon Leviathan on Watersday wearing Summoner's Doublet: got -3 day reduction value
Summon Titan on Watersday wearing Summoner's Doublet: got 0 day reduction value

File: src/map/utils/charutils.cpp [hunk 1]
```diff
@@ -5136,33 +5136,43 @@ namespace charutils
 
     uint16 AvatarPerpetuationReduction(CCharEntity* PChar)
     {
-        uint16 reduction = PChar->getMod(Mod::PERPETUATION_REDUCTION);
+        // REMEMBER:
```
Comment (claywar) [inline src/map/utils/charutils.cpp:5139]: Yess....!  Documentation!

**Review by claywar — APPROVED** [review]

**Comment by TeoTwawki** [comment]
> Don't really understand whats going on with the WEATHER_REDUCTION thing.

this morning my caffeine deprived brain was thinking that double weak weather was mistakenly identical to double strong where they were defined (I was looking at the old PR's diff), but beyond that a big "I dunno" from me as well.

**Comment by Skerxan** [comment]
Weather doesn't affect perp. cost unless you use Summoner's Horn and/or Caller's Pendant.
Corresponding weather with element of avatar activates it. Doesn't matter if single or double weather. Also, SCH can trigger it.

If that's what you mean.

Horn `15086`
Pendant `11619`
https://ffxiclopedia.fandom.com/wiki/Summoner's_Horn
https://ffxiclopedia.fandom.com/wiki/Caller%27s_Pendant

**Comment by zach2good** [comment]
> Weather doesn't affect perp. cost unless you use Summoner's Horn and/or Caller's Pendant. Corresponding weather with element of avatar activates it. Doesn't matter if single or double weather. Also, SCH can trigger it.
> 
> If that's what you mean.
> 
> Horn `15086` Pendant `11619` https://ffxiclopedia.fandom.com/wiki/Summoner's_Horn https://ffxiclopedia.fandom.com/wiki/Caller%27s_Pendant

Exactly the sort of thing I was looking for! I'll add in some more documentation for this tonight and then we're good to go

**Comment by zach2good** [comment]
Thoughts: The return type should be changed from `uint16` to `int16` -> is there a worst-case scenario where you summon Shiva but have some kickass HQ mega-fire staff out, and then the affinity bonus is so negative your reduction becomes negative?

Would love to try and get proof of this from retail

**Pull request merged**